openmc/weight_windows.py

```diff
@@ -410,3 +406,180 @@ def from_hdf5(cls, group, meshes):
+
+        # header value checks
+        if _if != 1:
+            raise ValueError(f'Found incorrect file type, if: {_if}')
+
+        if iv > 1:
+            # read number of time bins for each particle, 'nt(1...ni)'
+            nt = np.fromstring(wwinp.readline(), sep=' ', dtype=int)
+
+            # raise error if time bins are present for now
+            raise ValueError('Time-dependent weight windows '
+                             'are not yet supported')
+        else:
+            nt = ni * [1]
+
+        if nr == 16:
+            raise NotImplementedError('Cylindrical and spherical mesh '
+                                      'types are not yet supported')
+
+        # read number of energy bins for each particle, 'ne(1...ni)'
+        ne = np.fromstring(wwinp.readline(), sep=' ', dtype=int)
+
+        # read coarse mesh dimensions and lower left corner
+        mesh_description = np.fromstring(wwinp.readline(), sep=' ')
+        nfx, nfy, nfz = [int(x) for x in mesh_description[:3]]
+        xyz0 = mesh_description[3:]
+
+        # read cylindrical and spherical mesh vectors if present
+        if nr == 16:
+            # read number of coarse bins
+            line_arr = np.fromstring(wwinp.readline(), sep=' ')
+            ncx, ncy, ncz = [int(x) for x in line_arr[:3]]
+            # read polar vector (x1, y1, z1)
+            xyz1 = line_arr[3:] - xyz0
+            polar_vec = xyz1 / np.linalg.norm(xyz1)
+            # read azimuthal vector (x2, y2, z2)
+            line_arr = np.fromstring(wwinp.readline(), sep=' ')
+            xyz2 = line_arr[:3] - xyz0
+            azimuthal_vec = xyz2 / np.linalg.norm(xyz2)
+            # read geometry type
+            nwg = int(line_arr[-1])
+        elif nr == 10:
+            # read rectilinear data:
+            # number of coarse mesh bins and mesh type
+            ncx, ncy, ncz, nwg = \
+                [int(x) for x in np.fromstring(wwinp.readline(), sep=' ')]
+        else:
+            raise RuntimeError(f'Invalid mesh description (nr) found: {nr}')
+
+        # read BLOCK 2 and BLOCK 3 data into a single array
+        ww_data = np.fromstring(wwinp.read(), sep=' ')
+
+    # extract mesh data from the ww_data array
+    start_idx = 0
+
+    # first values in the mesh definition arrays are the first
+    # coordinate of the grid
+    end_idx = start_idx + 1 + 3 * ncx
+    x0, x_vals = ww_data[start_idx], ww_data[start_idx+1:end_idx]
+    start_idx = end_idx
+
+    end_idx = start_idx + 1 + 3 * ncy
+    y0, y_vals = ww_data[start_idx], ww_data[start_idx+1:end_idx]
+    start_idx = end_idx
+
+    end_idx = start_idx + 1 + 3 * ncz
+    z0, z_vals = ww_data[start_idx], ww_data[start_idx+1:end_idx]
+    start_idx = end_idx
+
+    # mesh consistency checks
+    if nr == 16 and nwg == 1:
+        raise ValueError(f'Mesh description in header ({nr}) '
+                         f'does not match the mesh type ({nwg})')
+
+    if not (xyz0 == (x0, y0, z0)).all():
```

Comment:
Slightly more concise
```suggestion
    if (xyz0 != (x0, y0, z0)).any():
```